src/server/controllers/index.ts

```diff
@@ -27,7 +27,7 @@ export function setupApiRoutes(app: express.Application): void {
   app.get('/api/users/me', authMiddleware, userController.getMe);
   app.post('/api/users/register', validate(userController.registerValidation), userController.registerUser);
   app.put('/api/users/resetpassword', validate(userController.resetPasswordValidation), userController.resetPassword);
-  app.put('/api/users/changepassword', authMiddleware, validate(userController.changePasswordValidation), userController.changePassword);
+  app.put('/api/users/changepassword', validate(userController.changePasswordValidation), userController.changePassword);
```

Comment:
Yep, second option is way better, it's more verbose. What if we call the endpoints:
- `/api/users/me/changepassword` - this would be the old `/api/users/changepassword` where ou have to be authenticated
AND
- `/api/users/{userId}/changepassword` - and here you have to pass also the correct `{password, resetPasswordId}`. 

I have a feeling it's more restful like this, what do you think?